### PR TITLE
update roles for member of team

### PR DESCRIPTION
**Pull request by dannitahdruk**

### CODE REVIEW

File: src/resources/utilities/constants.util.js [hunk 1]
```diff
@@ -345,6 +345,50 @@ const _searchDataTypes = {
...
+
+const _roleMemberTeam = {
+	CUST_TEAM_ADMIN: 'custodian.team.admin',
+	CUST_MD_MANAGER: 'custodian.metadata.manager',
+	CUST_MD_EDITOR: 'custodian.metadata.editor',
+	CUST_DEVELOPER: 'custodian.developer',
+	CUST_DAR_MANAGER: 'custodian.dar.manager',
+	CUST_DAR_REVIEWER: 'custodian.dar.reviewer',
+	CUST_DUR_MANAGER: 'custodian.dur.manager',
+};
```
Comment (kymmeh) [inline src/resources/utilities/constants.util.js:358]: @dannitahdruk we're only implementing the `custodian.team.admin`, `custodian.metadata.manager` and custodian.dar.manager` as part of this phase. The other roles are coming in a future phase.

Comment (kymmeh) [reply src/resources/utilities/constants.util.js:358]: https://hdruk.atlassian.net/wiki/spaces/HDT/pages/2241658883/IAM+Implementation+Approach

Comment (dannitahdruk) [reply src/resources/utilities/constants.util.js:358]: This part will be updated as required.

Comment (dannitahdruk) [reply src/resources/utilities/constants.util.js:364]: list was updated

Comment (kymmeh) [reply src/resources/utilities/constants.util.js:364]: What do you mean? We're only implementing these:

```
const _roleMemberTeam = {
	CUST_TEAM_ADMIN: 'custodian.team.admin',
	CUST_MD_MANAGER: 'custodian.metadata.manager',
	CUST_DAR_MANAGER: 'custodian.dar.manager',
};
```

Not these:

```
const _roleMemberTeam = {
	CUST_MD_EDITOR: 'custodian.metadata.editor',
	CUST_DEVELOPER: 'custodian.developer',
	CUST_DAR_REVIEWER: 'custodian.dar.reviewer',
	CUST_DUR_MANAGER: 'custodian.dur.manager',
	HDRUK_ADMIN: 'hdruk.admin',
	HDRUK_ONBOARDING_ADMIN: 'hdruk.onboarding.admin',
	HDRUK_METADATA_ADMIN: 'hdruk.metadata.admin',
	HDRUK_DAR_ADMIN: 'hdruk.dar.admin',
	HDRUK_DUR_ADMIN: 'hdruk.dur.admin',
	HDRUK_ENTITY_ADMIN: 'hdruk.entity.admin',
};
```

Comment (dannitahdruk) [reply src/resources/utilities/constants.util.js:356]: based on interface need to be like this:
```
const _roleMemberTeam = {
	CUST_TEAM_ADMIN: 'custodian.team.admin',
	CUST_MD_MANAGER: 'custodian.metadata.manager',
	CUST_MD_EDITOR: 'custodian.metadata.editor',
	CUST_DAR_MANAGER: 'custodian.dar.manager',
	CUST_DAR_REVIEWER: 'custodian.dar.reviewer',
};
```
5 roles not 3 roles
updated

Comment (kymmeh) [reply src/resources/utilities/constants.util.js:356]: The frontend is using:

```
const _roleMemberTeam = {
	CUST_TEAM_ADMIN: 'custodian.team.admin',
	CUST_MD_MANAGER: 'custodian.metadata.manager',
	CUST_MD_EDITOR: 'metadata_editor',
	CUST_DAR_MANAGER: 'custodian.dar.manager',
	CUST_DAR_REVIEWER: 'reviewer',
};
```

Comment (dannitahdruk) [reply src/resources/utilities/constants.util.js:356]: I updated but this is not a consistent way ...

Comment (kymmeh) [reply src/resources/utilities/constants.util.js:356]: I agree, but the work has been split into multiple phases.

File: src/resources/utilities/__mocks__/checkAllowNewRoles.mock.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const mockUserUpdateRolesFalse = [
+    "reviewer",
+    "custodian.team.admin",
+    "custodian.metadata.manager",
+    "custodian.dar.manager"
+];
+const mockAllowedRoles = [
+    "custodian.team.admin",
+    "custodian.metadata.manager",
+    "custodian.metadata.editor",
```
Comment (kymmeh) [inline src/resources/utilities/__mocks__/checkAllowNewRoles.mock.js:15]: These mocks just need updating.

**Pull request merged**